pkg/e2e/podplacement/pod_placement_test.go

```diff
@@ -254,6 +255,37 @@ var _ = Describe("The Pod Placement Operand", func() {
 			verifyPodNodeAffinity(ns, "app", "test")
 			verifyPodLabels(ns, "app", "test", e2e.Present, schedulingGateLabel)
 		})
+		It("should not set the node affinity when nodeName exist", func() {
+			var err error
+			By("Create an ephemeral namespace")
+			ns := framework.NewEphemeralNamespace()
+			err = client.Create(ctx, ns)
+			Expect(err).NotTo(HaveOccurred())
+			//nolint:errcheck
+			defer client.Delete(ctx, ns)
+			By("Get the name of a random worker node")
+			workerNodeName, err := framework.GetRandomNodeName(ctx, client, "node-role.kubernetes.io/worker", "")
+			Expect(workerNodeName).NotTo(BeEmpty())
+			Expect(err).NotTo(HaveOccurred())
+			By("Create a deployment using the container with nodeName existing")
+			ps := NewPodSpec().
+				WithContainersImages(helloOpenshiftPublicMultiarchImage).
+				WithNodeName(workerNodeName).
+				Build()
+			d := NewDeployment().
+				WithSelectorAndPodLabels(podLabel).
+				WithPodSpec(ps).
+				WithReplicas(utils.NewPtr(int32(1))).
+				WithName("test-deployment").
+				WithNamespace(ns.Name).
+				Build()
+			err = client.Create(ctx, &d)
+			Expect(err).NotTo(HaveOccurred())
+			By("Check PPO should not add schedulingGate label on pod and pod can be created and running")
+			verifyPodsAreRunning(ns, "app", "test")
+			verifyPodNodeAffinity(ns, "app", "test")
+			verifyPodLabels(ns, "app", "test", e2e.Absent, schedulingGateLabel)
```

Comment:
```suggestion
			By("The pod should not have been processed by the webhook and the scheduling gate label should not be added")
			verifyPodLabels(ns, "app", "test", e2e.Absent, schedulingGateLabel)
			By("The pod should keep the same node affinity provided by the users. No node affinity is added by the controller.")
			verifyPodNodeAffinity(ns, "app", "test")
			By("The pod should be running and not gated")
			verifyPodsAreRunning(ns, "app", "test")
```

I don't like them 100%, but having more detailed comments in the verification steps in particular is nice as it's easier to find which step is failing precisely.